File: mobile/ios/ShiftPlanner/ShiftPlanner/Features/Auth/Repositories/MockAuthRepository.swift
import Foundation

enum AuthError: LocalizedError {
    case emptyEmail
    case emptyPassword
    case emptyName
    case invalidEmail
    case passwordTooShort
    case passwordsDontMatch

    var errorDescription: String? {
        switch self {
        case .emptyEmail:
            return localized("Email is required.", "Почта обязательна.")
        case .emptyPassword:
            return localized("Password is required.", "Пароль обязателен.")
        case .emptyName:
            return localized("Name is required.", "Имя обязательно.")
        case .invalidEmail:
            return localized("Enter a valid email.", "Введите корректную почту.")
        case .passwordTooShort:
            return localized("Password must be at least 8 characters.", "Пароль должен содержать минимум 8 символов.")
            
        case .passwordsDontMatch:
            return localized("Passwords do not match.", "Пароли не совпадают.")
        }
    }
}

final class MockAuthRepository: AuthRepository {
    private var currentUser: AppUser?

    func login(email: String, password: String) async throws -> AppUser {
        try await Task.sleep(nanoseconds: 700_000_000)

        try validateLogin(email: email, password: password)

        let user = AppUser(
            id: UUID().uuidString,
            email: email,
            name: "Test User",
            role: .manager,
            employeeId: nil,
            company: nil,
            branch: nil,
            position: nil
        )

        currentUser = user
        return user
    }

    func signUp(
        email: String,
        password: String,
        name: String,
        role: UserRole
    ) async throws -> AppUser {
        try await Task.sleep(nanoseconds: 700_000_000)

        try validateSignUp(
            email: email,
            password: password,
            name: name
        )

        let user = AppUser(
            id: UUID().uuidString,
            email: email,
            name: name,
            role: role,
            employeeId: nil,
            company: nil,
            branch: nil,
            position: nil
        )

        currentUser = user
        return user
    }

    func logout() async {
        currentUser = nil
    }

    func deleteCurrentAccount() async throws {
        currentUser = nil
    }

    func getCurrentUser() async -> AppUser? {
        currentUser
    }

    private func validateLogin(email: String, password: String) throws {
        if email.isEmpty {
            throw AuthError.emptyEmail
        }

        if !email.contains("@") {
            throw AuthError.invalidEmail
        }

        if password.isEmpty {
            throw AuthError.emptyPassword
        }
    }

    private func validateSignUp(
        email: String,
        password: String,
        name: String
    ) throws {
        if name.isEmpty {
            throw AuthError.emptyName
        }

        if email.isEmpty {
            throw AuthError.emptyEmail
        }

        if !email.contains("@") {
            throw AuthError.invalidEmail
        }

        if password.isEmpty {
            throw AuthError.emptyPassword
        }

        if password.count < 8 {
            throw AuthError.passwordTooShort
        }
        
    }
}

File: mobile/android/app/src/main/java/com/froggyriia/shiftplanner/data/auth/MockAuthRepository.kt
package com.froggyriia.shiftplanner.data.auth

import com.froggyriia.shiftplanner.domain.model.AppUser
import com.froggyriia.shiftplanner.domain.model.UserRole
import kotlinx.coroutines.delay
import java.util.UUID

class MockAuthRepository : AuthRepository {
    private var currentUser: AppUser? = null

    override suspend fun login(
        email: String,
        password: String
    ): AppUser {
        delay(700)

        validateLogin(email, password)

        val user = AppUser(
            id = UUID.randomUUID().toString(),
            email = email,
            name = "Test User",
            role = UserRole.MANAGER
        )

        currentUser = user
        return user
    }

    override suspend fun signUp(
        email: String,
        password: String,
        name: String,
        role: UserRole
    ): AppUser {
        delay(700)

        validateSignUp(email, password, name)

        val user = AppUser(
            id = UUID.randomUUID().toString(),
            email = email,
            name = name,
            role = role
        )

        currentUser = user
        return user
    }

    override suspend fun logout() {
        currentUser = null
    }

    override suspend fun getCurrentUser(): AppUser? {
        return currentUser
    }

    private fun validateLogin(
        email: String,
        password: String
    ) {
        if (email.isBlank()) {
            throw IllegalArgumentException("Email is required")
        }

        if (!email.contains("@")) {
            throw IllegalArgumentException("Enter a valid email")
        }

        if (password.isBlank()) {
            throw IllegalArgumentException("Password is required")
        }
    }

    private fun validateSignUp(
        email: String,
        password: String,
        name: String
    ) {
        if (name.isBlank()) {
            throw IllegalArgumentException("Name is required")
        }

        if (email.isBlank()) {
            throw IllegalArgumentException("Email is required")
        }

        if (!email.contains("@")) {
            throw IllegalArgumentException("Enter a valid email")
        }

        if (password.isBlank()) {
            throw IllegalArgumentException("Password is required")
        }

        if (password.length < 6) {
            throw IllegalArgumentException("Password must be at least 6 characters")
        }
    }
}